### DISCUSSION TITLE
Create Github App from a manifest support in GitHub Enterprise

### DISCUSSION THREAD

**Discussion by fermayo** [community/community#24470] [Enterprise]
<p>Hello,</p>
<p>Are there any plans to add support for <a href="https://developer.github.com/apps/building-github-apps/creating-github-apps-from-a-manifest/" rel="nofollow noopener">creating GitHub Apps from manifest</a> to GitHub Enterprise? What is the current best practice for GitHub Enterprise users to install GitHub Apps?</p>
<p>Thanks,</p>
<p>Fernando.</p>

**Comment by lee-dohm** [comment]
<p>If you take a look at the document <a href="https://developer.github.com/apps/building-github-apps/creating-github-apps-from-a-manifest/#using-probot-to-implement-the-github-app-manifest-flow" rel="nofollow noopener">using Probot to implement the GitHub App manifest flow</a>, you can see how to create a GitHub App Manifest using Probot. And I know that <a href="https://github.com/probot/probot/issues/500#issuecomment-381266516" rel="nofollow noopener">Probot can be used with GitHub Enterprise</a>. So, I suspect this can be done with a couple extra steps, but I don’t know of a single-click App-to-Enterprise manifest system currently.</p>
<p>I hope that helps!</p>

**Comment by braeken** [comment]
<p>I have been spending quite some time to get some useful info on how our users would install a (probot based) github app that would work with with the GitHub Enterprise (server) that my team is responsible for.  And all the links go to <a href="https://github.com/probot/probot/issues/500" rel="nofollow noopener">https://github.com/probot/probot/issues/500 </a> which then points to <a href="https://probot.github.io/docs/deployment/" rel="nofollow noopener">https://probot.github.io/docs/deployment/ </a> but in my opinion it does not really answer the question especially because we need something that works within our firewalls.</p>
<p>All that we are looking for is some steps or suggestion on setting things up. I want to tell the users of our installation that if you want to use a probot based github app, then need to do\have …  but I don’t have an answer. At this moment I don’t even know if we, as the owner of the GitHub Enterprise installation need to do something to accomodate this.</p>
<p>Of course I can reach out to GitHub support but I believe that this is info that should be easily available for GitHub Enterprise Server users.</p>

**Comment by nadiajoyce** [comment]
<p>Hi there,</p>
<p>Thanks for sharing this information and your feedback. I can understand why that would be valuable. However, given the inevitable variance that occurs across different organization’s internal IT infrastructure, it would be very difficult to provide documentation that achieves what you’re asking for and is applicable to all organizations. This is more of a specific instance question, and I actually would recommend reaching out to <a href="https://enterprise.github.com/support" rel="nofollow noopener">GitHub Enterprise Support</a> for help.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>

**Comment by fermayo** [comment]
<blockquote>
<p><a class="mention" href="/nadiajoyce">@nadiajoyce</a> wrote:</p>
</blockquote>
<blockquote>
<p>However, given the inevitable variance that occurs across different organization’s internal IT infrastructure, it would be very difficult to provide documentation that achieves what you’re asking for and is applicable to all organizations. This is more of a specific instance question, and I actually would recommend reaching out to <a href="https://enterprise.github.com/support" rel="nofollow noopener">GitHub Enterprise Support</a> for help.</p>
</blockquote>
<p>I disagree. Implementing the “GitHub App from manifest workflow” in GitHub Enterprise as it is implemented in <a href="http://GitHub.com" rel="nofollow noopener">GitHub.com</a> is not specific to any instance. It is just a shortcut for the user of GitHub Enterprise to not having to fill in the “Create new GitHub App” form manually when integrating with software that provides a GitHub App. It has nothing to do with the organization’s internal IT infrastructure.</p>

**Comment by lee-dohm** [comment]
<p>The question that <a class="mention" href="/braeken">@braeken</a> asked was specifically referencing the Probot deployment documentation and how to achieve that inside their organization’s firewalls. How to deploy something within any given organization’s firewalls has everything to do with that organization’s IT infrastructure and isn’t something that we can answer.</p>

**Comment by fermayo** [comment]
<p>Yes, you are right, the reply by <a class="mention" href="/nadiajoyce">@nadiajoyce</a> wasn’t for my question. My bad.</p>

**Comment by tcbyrd** [comment, marked as answer]
<p>Just wanted to add that app manifests are a fairly new feature, so this flow became available on the most recent release of <a href="https://developer.github.com/enterprise/2.16/apps/building-github-apps/creating-github-apps-from-a-manifest/" rel="nofollow noopener">GitHub Enterprise (2.16)</a>. Previous to 2.16 you have to <a href="https://developer.github.com/enterprise/2.16/apps/building-github-apps/creating-a-github-app/" rel="nofollow noopener">create the GitHub App manually</a>. Once you add all the environment variables (App ID &amp; Private Key), Probot will effectively “skip” the manifest setup option.</p>
